globals/matchmaking_utils.go

```diff
@@ -292,6 +347,181 @@ func AddPlayersToSession(session *CommonMatchmakeSession, connectionIDs []uint32
+			if target == nil {
+				// TODO - Error here?
+				//logger.Warning("Player not found")
+				continue
+			}
+
+			notificationRequestMessage := nex.NewRMCRequest()
+			notificationRequestMessage.SetProtocolID(notifications.ProtocolID)
+			notificationRequestMessage.SetCallID(CurrentMatchmakingCallID.Increment())
+			notificationRequestMessage.SetMethodID(notifications.MethodProcessNotificationEvent)
+
+			notificationCategory := notifications.NotificationCategories.Participation
+			notificationSubtype := notifications.NotificationSubTypes.Participation.NewParticipant
+
+			oEvent := notifications_types.NewNotificationEvent()
+			oEvent.PIDSource = initiatingClient.PID()
+			oEvent.Type = notifications.BuildNotificationType(notificationCategory, notificationSubtype)
+			oEvent.Param1 = session.GameMatchmakeSession.ID
+			oEvent.Param2 = target.PID()
+			oEvent.StrParam = joinMessage
+			oEvent.Param3 = uint32(len(connectionIDs))
+
+			notificationStream := nex.NewStreamOut(server)
+
+			notificationStream.WriteStructure(oEvent)
+
+			notificationRequestMessage.SetParameters(notificationStream.Bytes())
+			notificationRequestBytes := notificationRequestMessage.Bytes()
+
+			var messagePacket nex.PacketInterface
+
+			if server.PRUDPVersion() == 0 {
+				messagePacket, _ = nex.NewPacketV0(initiatingClient, nil)
+				messagePacket.SetVersion(0)
+			} else {
+				messagePacket, _ = nex.NewPacketV1(initiatingClient, nil)
+				messagePacket.SetVersion(1)
+			}
+
+			messagePacket.SetSource(0xA1)
+			messagePacket.SetDestination(0xAF)
+			messagePacket.SetType(nex.DataPacket)
+			messagePacket.SetPayload(notificationRequestBytes)
+
+			messagePacket.AddFlag(nex.FlagNeedsAck)
+			messagePacket.AddFlag(nex.FlagReliable)
+
+			server.Send(messagePacket)
+		}
+
+		notificationRequestMessage := nex.NewRMCRequest()
+		notificationRequestMessage.SetProtocolID(notifications.ProtocolID)
+		notificationRequestMessage.SetCallID(CurrentMatchmakingCallID.Increment())
+		notificationRequestMessage.SetMethodID(notifications.MethodProcessNotificationEvent)
+
+		notificationCategory := notifications.NotificationCategories.Participation
+		notificationSubtype := notifications.NotificationSubTypes.Participation.NewParticipant
+
+		oEvent := notifications_types.NewNotificationEvent()
+		oEvent.PIDSource = initiatingClient.PID()
+		oEvent.Type = notifications.BuildNotificationType(notificationCategory, notificationSubtype)
+		oEvent.Param1 = session.GameMatchmakeSession.ID
+		oEvent.Param2 = initiatingClient.PID()
+		oEvent.StrParam = joinMessage
+		oEvent.Param3 = uint32(len(connectionIDs))
+
+		notificationStream := nex.NewStreamOut(server)
+
+		notificationStream.WriteStructure(oEvent)
+
+		notificationRequestMessage.SetParameters(notificationStream.Bytes())
+		notificationRequestBytes := notificationRequestMessage.Bytes()
+
+		target := server.FindClientFromPID(uint32(session.GameMatchmakeSession.Gathering.OwnerPID))
```

Comment:
We have to check if `target` is nil, like we are doing on the other notifications